### PR TITLE
Updates to git tree listing utilities

### PR DESCRIPTION
**Pull request by hoytak**
This update adds API surface area to the git tree listing functionality.  This allows more ability to query a repository for additional tree functionality -- something that is needed server side.

### CODE REVIEW

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs [hunk 1]
```diff
@@ -45,6 +44,21 @@ pub fn repo_dir_from_repo(repo: &Arc<Repository>) -> PathBuf {
     .to_path_buf()
 }
 
+pub fn normalize_repo_path(path: &str) -> String {
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:47]: Can we add a comment to describe the intent of this function?

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs [hunk 2]
```diff
@@ -269,23 +290,73 @@ pub fn read_file_from_repo(
...
+    pub fn from_tree_entry(
+        repo: &Repository,
+        prefix: &str,
+        entry: &git2::TreeEntry<'_>,
+    ) -> Option<Self> {
+        let file_name = entry.name().unwrap().to_owned();
+        let file_path = Path::new(prefix).join(&file_name);
+        let name = file_path.to_str().unwrap().to_owned();
+
+        eprintln!(
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:312]: maybe lower this to debug!

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs [hunk 3]
```diff
@@ -269,23 +290,73 @@ pub fn read_file_from_repo(
...
+            "prefix={prefix}, file_name = {file_name:?}, file_path={file_path:?},name={name}"
+        );
+
+        match entry.kind() {
+            Some(git2::ObjectType::Blob) => {
+                let size = entry.to_object(repo).unwrap().as_blob().unwrap().size();
+                Some(ListObjectEntry {
+                    path: name,
+                    oid: entry.id(),
+                    size,
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:322]: Should we show the smudged size?

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs [hunk 4]
```diff
@@ -295,62 +366,122 @@ pub fn list_files_from_repo(
...
+                        (tree, "")
+                    }
+                }
+                Some(ObjectType::Blob) => {
+                    // this is a single file, directly return it
+                    let size = entry.to_object(repo).unwrap().as_blob().unwrap().size();
+                    let entry = ListObjectEntry {
+                        path: prefix.to_owned(),
+                        oid: entry.id(),
+                        size,
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:395]: Also this size

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs [hunk 5]
```diff
@@ -295,62 +366,122 @@ pub fn list_files_from_repo(
...
+    let mut entries = Vec::new();
+
+    for entry in tree.iter() {
+        let object = entry.to_object(repo)?;
+
+        let (size, is_tree) = match object.kind() {
+            Some(ObjectType::Blob) => (object.as_blob().unwrap().size(), false),
+            Some(ObjectType::Tree) => (0, true),
+            _ => {
+                warn!("Git object entry {object:?} in tree {tree:?} has type other than blob or tree.");
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:451]: maybe lower to debug!

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs [hunk 6]
```diff
@@ -295,62 +366,122 @@ pub fn list_files_from_repo(
...
+) -> Result<Vec<ListObjectEntry>> {
+    // Returns a sorted list of dir entries, either blobs or directories, given by the tree oid
+    let tree = repo.find_tree(tree_oid)?;
+    let mut entries = Vec::new();
+
+    for entry in tree.iter() {
+        let object = entry.to_object(repo)?;
+
+        let (size, is_tree) = match object.kind() {
+            Some(ObjectType::Blob) => (object.as_blob().unwrap().size(), false),
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:448]: Also this size

**Review by seanses — COMMENTED** [review]
In general looks good to me. There's a concern over returning the cleaned file size. Since this is to mimic Xetea/s3 behavior I think we should return the smudged file size.

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs (hunk 2 above)
Comment (hoytak) [reply rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:312]: Ooops!  Debugging code that snuck in.

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs (hunk 5 above)
Comment (hoytak) [reply rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:451]: In this case it should never happen...   So something weird is going on if this happens.

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs (hunk 1 above)
Comment (hoytak) [reply rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:47]: Sure.

**Review by seanses — APPROVED** [review]

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs (hunk 5 above)
Comment (seanses) [reply rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:450]: Ah yeah warn! is fine

File: rust/gitxetcore/src/git_integration/git_repo_plumbing.rs (hunk 3 above)
Comment (hoytak) [reply rust/gitxetcore/src/git_integration/git_repo_plumbing.rs:321]: Let's do that at a higher level.

**Pull request merged**